packages/openlogin-utils/src/whitelabel.ts

```diff
@@ -0,0 +1,49 @@
+import Color from "color";
+
+import { type WHITE_LABEL_THEME } from "./interfaces";
+
+function getColorsList(colorsAmount = 3, colorsShiftAmount = 50, mixColor = "black", rotate = 0, saturation = 20, mainColor = "#0346ff") {
+  const colorsList = [];
+
+  let step;
+  for (step = 0; step < colorsAmount; step += 1) {
+    colorsList.push(
+      Color(mainColor)
+        .rotate(((step + 1) / colorsAmount) * -rotate)
+        .saturate(((step + 1) / colorsAmount) * (saturation / 100))
+        .mix(Color(mixColor), ((colorsShiftAmount / 100) * (step + 1)) / colorsAmount)
+        .hex()
+    );
+  }
+
+  return colorsList;
+}
+
+function generateWhiteLabelTheme(primary: string) {
+  const darkSet = getColorsList(3, 50, "black", 0, 20, primary);
+  const lightSet = getColorsList(6, 85, "white", 0, 20, primary);
+  return [...darkSet.reverse(), ...[primary], ...lightSet];
+}
+
+function applyWhiteLabelTheme(rootElement: HTMLElement, theme: WHITE_LABEL_THEME) {
```

Comment:
1. Does it make sense to put this function in openlogin-utils? Will it affect build size? If so can we code split it?
2. should this function instead only return theme colors and we let the caller of the function apply the colors themselves?
Technically the only colors that would need to be returned would be the primary color palette as the other colors don't require other shades being generated by us.

Reply:
1. i think it's fine if we will decide to put Whitelabel (source of truth like  here interface etc..). controllers should get WhiteLabel interface here as well. not sure about the size and code splitting but i think it's fine.
2. it's fine to apply the colors here. since should be using same css variables for colors. i think it should not be limited to primary to be more scalable but since we are only passing primary, then it will only process primary
2a. we can still return the color list in case the app using this util wants to apply it or use it for something else